sdk/cognitivelanguage/ai-language-text/review/ai-language-text.api.md

```diff
@@ -142,14 +179,15 @@ export interface BatchActionState<Kind extends AnalyzeBatchActionName> {
 }
 
 // @public
-export interface BatchActionSuccessResult<T, Kind extends AnalyzeBatchActionName> extends BatchActionState<Kind> {
+export interface BatchActionSuccessResult<T extends DocumentDetectedLanguage, Kind extends AnalyzeBatchActionName> extends BatchActionState<Kind> {
     readonly completedOn: Date;
     readonly error?: undefined;
     readonly results: T[];
 }
 
 // @public
 export interface BeginAnalyzeBatchOptions extends TextAnalysisOperationOptions {
+    defaultLanguage?: string;
```

Comment:
Does JS provide a defaultLanguage option at the client-level? We do in Python. This gets passed as the language hint per document. Need to think through how it will coexist with this parameter.

Reply:
Excellent point, I think the problem here is that the client defaults the client-level hint to "en" so the client will currently always send records with hints. I believe the new option should override the default and the requests shouldn't contain any hints in this case.

The interesting question is which mechanism should have precedence when the client has a set hint, but the call also requires auto LD? or just error?

Reply:
Opened an issue here: https://github.com/Azure/azure-sdk-for-python/issues/26176

Yeah need to think this through. I think I'll create a table of all possibilities. Seems auto detection is opt-in from what I can tell: https://github.com/Azure/azure-rest-api-specs/blob/main/dev/cognitiveservices/data-plane/Language/analyzetext.json#L2855

Reply:
I addressed this in https://github.com/Azure/azure-sdk-for-js/pull/23208/commits/3d7138280ff9710d5dd8415f19d87c68e712809f by making auto LD take precedence if actually used. I can revisit this later if a better approach emerges.

Reply:
Okay took more than a minute to look at this and realized that "auto" is passed at a per-document level. So I agree with your approach and think this will coexist nicely with our `default_language` at the client-level and our `language` at the client-method-level since these two options are just a short-cut for setting a per-document language for the user. My one thought is that we might want to rename the service's new [default_language](https://github.com/Azure/azure-rest-api-specs/blob/main/dev/cognitiveservices/data-plane/Language/analyzetext.json#L517) to something that will make more clear that this is the fall back language used IF autodetection is selected and the service fails to detect the language. Right now I think the name of this parameter is kind of confusing in context of what we already have in the client libraries.

Reply:
Good point on naming, I opened https://github.com/Azure/azure-sdk-for-js/issues/23239 to track this. I reluctant to address it for this beta though due to time constraints.